commit 3b2164b588b097dcf0cf5154b3580d63493d8d46
Author: Chris Friedt <cfriedt@tenstorrent.com>
Date:   Sun Nov 23 12:35:33 2025 -0500

    net: sockets: prefix posix flags with ZVFS_
    
    Prefix internal constants with ZVFS_ in the socket and eventfd areas.
    
    Signed-off-by: Chris Friedt <cfriedt@tenstorrent.com>

diff --git a/lib/os/zvfs/zvfs_eventfd.c b/lib/os/zvfs/zvfs_eventfd.c
index bdd93b10a9a..6c245ad11bd 100644
--- a/lib/os/zvfs/zvfs_eventfd.c
+++ b/lib/os/zvfs/zvfs_eventfd.c
@@ -235,11 +235,11 @@ static int zvfs_eventfd_ioctl_op(void *obj, unsigned int request, va_list args)
 	}
 
 	switch (request) {
-	case F_GETFL:
+	case ZVFS_F_GETFL:
 		ret = efd->flags & ZVFS_EFD_FLAGS_SET;
 		break;
 
-	case F_SETFL: {
+	case ZVFS_F_SETFL: {
 		int flags;
 
 		flags = va_arg(args, int);
diff --git a/subsys/net/lib/sockets/socketpair.c b/subsys/net/lib/sockets/socketpair.c
index b0de6e72385..2d6e59d47c3 100644
--- a/subsys/net/lib/sockets/socketpair.c
+++ b/subsys/net/lib/sockets/socketpair.c
@@ -929,19 +929,19 @@ static int spair_ioctl(void *obj, unsigned int request, va_list args)
 	have_local_sem = true;
 
 	switch (request) {
-		case F_GETFL: {
-			if (sock_is_nonblock(spair)) {
-				flags |= O_NONBLOCK;
-			}
-
-			res = flags;
-			goto out;
+	case ZVFS_F_GETFL: {
+		if (sock_is_nonblock(spair)) {
+			flags |= ZVFS_O_NONBLOCK;
 		}
 
-		case F_SETFL: {
+		res = flags;
+		goto out;
+	}
+
+		case ZVFS_F_SETFL: {
 			flags = va_arg(args, int);
 
-			if (flags & O_NONBLOCK) {
+			if (flags & ZVFS_O_NONBLOCK) {
 				spair->flags |= SPAIR_FLAG_NONBLOCK;
 			} else {
 				spair->flags &= ~SPAIR_FLAG_NONBLOCK;
diff --git a/subsys/net/lib/sockets/sockets_inet.c b/subsys/net/lib/sockets/sockets_inet.c
index a9998d8e15d..73106aed06e 100644
--- a/subsys/net/lib/sockets/sockets_inet.c
+++ b/subsys/net/lib/sockets/sockets_inet.c
@@ -3045,19 +3045,19 @@ static int sock_ioctl_vmeth(void *obj, unsigned int request, va_list args)
 	switch (request) {
 
 	/* In Zephyr, fcntl() is just an alias of ioctl(). */
-	case F_GETFL:
+	case ZVFS_F_GETFL:
 		if (sock_is_nonblock(obj)) {
-			return O_NONBLOCK;
+			return ZVFS_O_NONBLOCK;
 		}
 
 		return 0;
 
-	case F_SETFL: {
+	case ZVFS_F_SETFL: {
 		int flags;
 
 		flags = va_arg(args, int);
 
-		if (flags & O_NONBLOCK) {
+		if (flags & ZVFS_O_NONBLOCK) {
 			sock_set_flag(obj, SOCK_NONBLOCK, SOCK_NONBLOCK);
 		} else {
 			sock_set_flag(obj, SOCK_NONBLOCK, 0);
diff --git a/include/zephyr/sys/fdtable.h b/include/zephyr/sys/fdtable.h
index 676c0dc3a37..a5c93350089 100644
--- a/include/zephyr/sys/fdtable.h
+++ b/include/zephyr/sys/fdtable.h
@@ -49,6 +49,22 @@
 #define ZVFS_MODE_IFLNK  0120000
 #define ZVFS_MODE_IFSOCK 0140000
 
+#define ZVFS_S_IRWXU 00700
+#define ZVFS_S_IRUSR 00400
+#define ZVFS_S_IWUSR 00200
+#define ZVFS_S_IXUSR 00100
+#define ZVFS_S_IRWXG 00070
+#define ZVFS_S_IRGRP 00040
+#define ZVFS_S_IWGRP 00020
+#define ZVFS_S_IXGRP 00010
+#define ZVFS_S_IRWXO 00007
+#define ZVFS_S_IROTH 00004
+#define ZVFS_S_IWOTH 00002
+#define ZVFS_S_IXOTH 00001
+#define ZVFS_S_ISUID 04000
+#define ZVFS_S_ISGID 02000
+#define ZVFS_S_ISVTX 01000
+
 #define ZVFS_POLLIN   BIT(0)
 #define ZVFS_POLLPRI  BIT(1)
 #define ZVFS_POLLOUT  BIT(2)
